### ISSUE TITLE
Non-manifold fluid meshes

### ISSUE THREAD

**Issue by DaneHoffman** [rlguy/Blender-FLIP-Fluids#622]
#### System Information

Blender Version: 3.6.1, blender-v3.6-release, 2023-07-17 12:50, 8bda729ef4dc
Addon Version: A FLIP Fluid Simulation Tool for Blender (v1.7.0 Stable 27-JUN-2023)
OS: Windows-10-10.0.22621-SP0
GPU: NVIDIA GeForce RTX 2080 Ti/PCIe/SSE2 NVIDIA Corporation 4.5.0 NVIDIA 545.37
CPU: AMD Ryzen 9 3900X 12-Core Processor            
CPU Threads: 24
RAM: 32GB

#### Describe the bug

Large holes are appearing in the surface of the fluid mesh. Seemingly happens on faces that are close to perfectly horizontal. 

#### To Reproduce

Download .blend file. Point FLIP fluids addon to fluid cache directory, go to frame 253, and look at the bottom fluid inflow object. The top of the fluid mesh is missing. Problem seemingly becomes worse (holes become larger) at higher resolutions. Interestingly, happens with both the FLIP solver and the APIC solver. 

#### Expected Behaviour

Fluid meshes should be manifold geometry

#### Actual Behaviour

Fluid meshes are non-manifold at intermittent frames, have sharp edges and no volume. 

#### Screenshots

![Screenshot 2023-07-22 142726](https://github.com/rlguy/Blender-FLIP-Fluids/assets/35747100/e64e618c-26ad-4c25-996e-6a767c90969d)

Blend file:
[flip-fluids-mesh-bug.zip](https://github.com/rlguy/Blender-FLIP-Fluids/files/12137259/flip-fluids-mesh-bug.zip)

Cache file:
[fluidcache.zip](https://github.com/rlguy/Blender-FLIP-Fluids/files/12137262/fluidcache.zip)

**Comment by rlguy** [comment]
Hi, thanks for the report! I am able to see the issue in the attached cache and am also able to reproduce the issue on the same frames when baking the simulation.

I should be able to look further into this beginning the week of July 24th and will keep you updated. We are releasing a new FLIP Fluids version 1.7.1 during this week, so hopefully we'll be able to fix this issue for that version.

If you need an immediate workaround, running the following script within the Blend file should work. This script will use more RAM resources, so it is not an ideal workaround.

```python
import bpy
domain_properties = bpy.context.scene.flip_fluid.get_domain_properties()
domain_properties.surface.compute_chunk_mode = 'COMPUTE_CHUNK_MODE_FIXED'
domain_properties.surface.compute_chunks_fixed = 1
```

Attached is the Blend file including the script: [flip-fluids-mesh-bug-revised.zip](https://github.com/rlguy/Blender-FLIP-Fluids/files/12137566/flip-fluids-mesh-bug-revised.zip)

![image](https://github.com/rlguy/Blender-FLIP-Fluids/assets/4285191/aa128e19-c86e-4bf4-80ca-a52c61b71952)

If you want to revert the Blend file back to the default behavior, this script can be run:

```python
import bpy
domain_properties = bpy.context.scene.flip_fluid.get_domain_properties()
domain_properties.surface.compute_chunk_mode = 'COMPUTE_CHUNK_MODE_AUTO'
```

**Comment by rlguy** [comment]
I just had a chance to investigate the issue and it looks like this bug went unnoticed or unreported since June 2019. Thanks again for the report!

This bug was introduced due to an optimization that was added to the mesh generator. A bug in the optimization caused important mesh data to be discarded incorrectly resulting in missing triangles. It looks like this bug can happen when particle positions are relatively flat on the X, Y, or Z axis plane (specifically the plane perpendicular to the longest side of the domain) and are located in specific positions along that axis within the domain.

This issue has been fixed for FLIP Fluids 1.7.1 which will be released tomorrow on July 25th, 2023.
